keras_cv/losses/smooth_l1.py

```diff
@@ -19,33 +19,33 @@
 class SmoothL1Loss(tf.keras.losses.Loss):
     """Implements Smooth L1 loss.
 
-    SmoothL1Loss implements the SmoothL1 function, where values less than `l1_cutoff`
+    SmoothL1Loss implements the SmoothL1 function, where values less than `beta`
     contribute to the overall loss based on their squared difference, and values greater
-    than l1_cutoff contribute based on their raw difference.
+    than `beta` contribute based on their raw difference.
 
     Args:
-        l1_cutoff: differences between y_true and y_pred that are larger than `l1_cutoff` are
+        beta: differences between y_true and y_pred that are larger than `beta` are
             treated as `L1` values
     """
 
-    def __init__(self, l1_cutoff=1.0, **kwargs):
+    def __init__(self, beta=1.0, **kwargs):
         super().__init__(**kwargs)
-        self.l1_cutoff = l1_cutoff
+        self.beta = beta
 
     def call(self, y_true, y_pred):
         difference = y_true - y_pred
         absolute_difference = tf.abs(difference)
         squared_difference = difference**2
         loss = tf.where(
-            absolute_difference < self.l1_cutoff,
-            0.5 * squared_difference,
-            absolute_difference - 0.5,
+            absolute_difference < self.beta,
+            (0.5 * squared_difference) / self.beta,
```

Comment:
ah I see that this actually changes the numerics

Reply:
Yup - the original name was a clear use of the argument, since it was just a cutoff. It works perfectly the same as the change when the cutoff is 1.0 (default), but as soon as it changes, the loss was technically incorrect.
We can still keep the cutoff name for backwards compatability if you think that's a good idea, but I think we'd benefit more from having it change early on to a slightly more accurate term?

Reply:
Reference formulas:

- https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html
- https://paperswithcode.com/method/self-adjusting-smooth-l1-loss

Reply:
We should actually lose the parentheses in the second statement. Updating in a moment

Reply:
Updated and updated docs to make the effect of `beta` clearer!

Reply:
Pinging since I think that the fix for the loss func is relevant for the 0.4 release @LukeWood 
Not blocking by any means, but worth having if we're releasing OD/SemSeg